extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -252,9 +252,38 @@ function createDicomWebApi(dicomWebConfig, UserAuthenticationService) {
         sortFunction
       );
 
+      /**
+       * naturalizes the dataset, and adds a retrieve bulkdata method
+       * to any values containing BulkDataURI.
+       * @param {*} instance
+       * @returns naturalized dataset, with retrieveBulkData methods
+       */
+      const addRetrieveBulkData = instance => {
```

Comment:
@swederik and @sedghi  Could you take a look at how this works?  There is an async function that gets added which gets called when the actual value is needed.  That should support both Overlay and Palette colour data, as well as anything else that happens to get added as BulkDataURI.  It isn't perfect since it requires the user of the data to know whether to call it or not, but since it is the user who knows whether they need that data, I couldn't see a better location for tracking it.